great_expectations/core/factory/checkpoint_factory.py

```diff
@@ -84,6 +84,12 @@ def get(self, name: str) -> Checkpoint:
 
         return self._get(key=key)
 
+    @public_api
+    @override
+    def all(self) -> List[Checkpoint]:
```

Comment:
As long as we are starting our can we make the initial interface for these `all` method more generic.
`typing.Iterable[Checkpoint]` or `typing.Sequence[Checkpoint]`.

These will free use up to do things like implement some of these `.add()` methods as lazy loading generator functions.
```suggestion
    def all(self) -> Iterator[Checkpoint]:
```

Reply:
Glad your brought this up! My initial plan with this was make it an `Iterable` (that's what the ticket even specifies), but I went with a `List` because I figure users might want to iterate through it multiple times, etc. Agreed that `Sequence` is preferable to `List`. I'll at least change it to that, but I'm open to arguments about preferring `Iterable`.

Reply:
The implementation doesn't need to change at all, you can still return a list.
It just gives us more flexibility going forward.

Reply:
```python
from collections.abc import Iterable
from string import ascii_letters
from typing import Protocol, runtime_checkable


@runtime_checkable
class SupportsAll(Protocol):
    def all(self) -> Iterable: ...


class ListAll:
    def all(self) -> Iterable[str]:
        return list(ascii_letters)


class GeneratorAll:
    def all(self) -> Iterable[str]:
        yield from ascii_letters


class StrAll:
    def all(self) -> Iterable[str]:
        return ascii_letters


l = ListAll()
g = GeneratorAll()
s = StrAll()

assert isinstance(l, SupportsAll)
assert isinstance(g, SupportsAll)
assert isinstance(s, SupportsAll)

l_all = l.all()
g_all = g.all()
s_all = s.all()

print(f"{type(l_all)} {l_all=}")
print(f"{type(g_all)} {g_all=}")
print(f"{type(s_all)} {s_all=}")

assert isinstance(l_all, Iterable)
assert isinstance(l_all, Iterable)
assert isinstance(l_all, Iterable)

assert list(l_all) == list(g_all) == list(s_all) == list(ascii_letters)
```